modules/adb-lakehouse/azure_data_factory.tf

```diff
@@ -1,6 +1,8 @@
 resource "azurerm_data_factory" "adf" {
+  count               = var.data_factory_name != "" ? 1 : 0
```

Comment:
yes, it works fine both when I define a data_factory_name variable in tfvars, and when I keep the default (`""`).

I can switch to   count               = var.data_factory_name == "" ? 0 : 1, if you think that it's more readable.

The only drawback of this is that because it's in a module, you still need to pass this variable, and users would need to know that `""` disables ADF.